src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java

```diff
@@ -0,0 +1,161 @@
+        if (Strings.isNullOrEmpty(caseTypeId)) {
+            throw new BadSearchRequest("Case type ID is required.");
+        }
+
+        JsonNode searchRequest = validateAndConvertRequest(jsonSearchRequest);
+        applyConfiguredSort(searchRequest, caseTypeId, useCase);
+
+        return new CrossCaseTypeSearchRequest.Builder()
+            .withCaseTypes(Collections.singletonList(caseTypeId))
+            .withSearchRequest(searchRequest)
+            .build();
+    }
+
+    private void applyConfiguredSort(JsonNode searchRequest, String caseTypeId, String useCase) {
+        JsonNode sortNode = searchRequest.get(SORT);
+        if (sortNode == null || useCase == null) {
+            ArrayNode appliedSortsNode = buildSortNode(caseTypeId, useCase);
+            if (appliedSortsNode.size() > 0) {
+                ((ObjectNode)searchRequest).set(SORT, appliedSortsNode);
+            }
+        }
+    }
+
+    private ArrayNode buildSortNode(String caseTypeId, String useCase) {
+        ArrayNode sortNode = objectMapper.createArrayNode();
+        addCaseTypeSorts(caseTypeId, useCase, sortNode);
+        return sortNode;
+    }
+
+    private void addCaseTypeSorts(String caseTypeId, String useCase, ArrayNode sortNode) {
+        Optional<CaseTypeDefinition> caseTypeOpt = getCaseTypeOperation.execute(caseTypeId, CAN_READ);
+        caseTypeOpt.ifPresent(caseType -> searchQueryOperation.getSortOrders(caseType, useCase)
+            .forEach(field -> sortNode.add(buildSortOrderFieldNode(caseType, field))));
+    }
+
+    private ObjectNode buildSortOrderFieldNode(CaseTypeDefinition caseTypeDefinition, SortOrderField sortOrderField) {
+        ObjectNode objectNode = objectMapper.createObjectNode();
+        CommonField commonField = caseTypeDefinition.getComplexSubfieldDefinitionByPath(sortOrderField.getCaseFieldId()).orElseThrow(() ->
+            new NullPointerException(String.format("Case field '%s' does not exist in configuration for case type '%s'.",
+                sortOrderField.getCaseFieldId(), caseTypeDefinition.getId()))
+        );
+        FieldTypeDefinition fieldType = commonField.getFieldTypeDefinition();
+
+        StringBuilder sb = new StringBuilder();
+
+        if (sortOrderField.isMetadata()) {
+            sb.append(MetaData.CaseField.valueOfReference(sortOrderField.getCaseFieldId()).getDbColumnName());
+        } else {
+            sb.append(CASE_DATA_PREFIX).append(sortOrderField.getCaseFieldId());
+            if (fieldType.getType().equals(FieldTypeDefinition.COLLECTION)) {
+                sb.append(COLLECTION_VALUE_SUFFIX);
+            }
+        }
+
+        if ((sortOrderField.isMetadata() && elasticsearchMappings.isDefaultTextMetadata(sb.toString()))
+             || (!sortOrderField.isMetadata() && elasticsearchMappings.isDefaultTextCaseData(fieldType))) {
+            sb.append(KEYWORD_SUFFIX);
```

Comment:
it would be good adding some explanatory comment here since it's not possible to deduct from the code why we are doing this.